src/plugin.rs

```diff
@@ -50,12 +56,32 @@ impl Serialize for Error {
 
 type Result<T> = std::result::Result<T, Error>;
 
+#[cfg(feature = "sqlite")]
+fn get_app_path<R: Runtime>(app: &AppHandle<R>) -> String {
+    app.path_resolver()
+        .app_dir()
+        .unwrap()
+        .as_os_str()
+        .to_str()
+        .unwrap()
+        .to_string()
+}
```

Comment:
So what's interesting is that pulling off the AppHandle directly from a command was my original intent but then the pattern in this repo seemed to prefer accessing things via the State plugin. More importantly when I first looked at the signatures of the commands in this plugin I didn't understand why the first parameter passed in _wasn't_ the AppHandle (as it was in other examples).

I have now come to believe that the **tauri::command** macro greedly consumes parameters  that it can match. Is that correct?

Reply:
WRT to the "too many conversions" ... I think I got convinced a String would be easier for me and this seemed like the path to this ambition but yeah I can see that we'd be better off with just the PathBuf. I'm just not used to distinguishing between strings and paths I guess :)

Reply:
> I have now come to believe that the tauri::command macro greedly consumes parameters that it can match. Is that correct?

Hm, i don't know about the "greedily" part, but yes, it checks the type of each argument and injects stuff _if needed_/asked for (`Window`, `AppHandle`, `State<>`)